x/crosschain/keeper/verify_block_header.go

```diff
@@ -0,0 +1,56 @@
+package keeper
+
+import (
+	"fmt"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	eth "github.com/ethereum/go-ethereum/common"
+	ethtypes "github.com/ethereum/go-ethereum/core/types"
+	"github.com/zeta-chain/zetacore/common"
+	"github.com/zeta-chain/zetacore/x/crosschain/types"
+)
+
+func (k Keeper) VerifyProof(ctx sdk.Context, proof *common.Proof, hash string, txIndex int64, chainID int64) (ethtypes.Transaction, error) {
+	var txx ethtypes.Transaction
+	crosschainFlags, found := k.zetaObserverKeeper.GetCrosschainFlags(ctx)
+	if !found {
+		return txx, fmt.Errorf("crosschain flags not found")
+	}
+	if crosschainFlags.BlockHeaderVerificationFlags == nil {
+		return txx, fmt.Errorf("block header verification flags not found")
+	}
+	if common.IsBitcoinChain(chainID) && !crosschainFlags.BlockHeaderVerificationFlags.IsBtcTypeChainEnabled {
+		return txx, fmt.Errorf("cannot verify proof for bitcoin chain %d", chainID)
+	}
+
+	if common.IsEVMChain(chainID) && !crosschainFlags.BlockHeaderVerificationFlags.IsEthTypeChainEnabled {
+		return txx, fmt.Errorf("cannot verify proof for evm chain %d ", chainID)
+	}
+
+	senderChain := common.GetChainFromChainID(chainID)
+	if senderChain == nil {
+		return txx, types.ErrUnsupportedChain
+	}
+
+	if !senderChain.IsProvable() {
+		return txx, fmt.Errorf("chain %d does not support block header verification", chainID)
+	}
+
+	blockHash := eth.HexToHash(hash)
+
+	res, found := k.zetaObserverKeeper.GetBlockHeader(ctx, blockHash.Bytes())
+	if !found {
+		return txx, fmt.Errorf("block header not found %s", blockHash)
+	}
+
+	// verify and process the proof
+	val, err := proof.Verify(res.Header, int(txIndex))
```

Comment:
It's better for this return `val` ([]byte) directly to make to more general as bitcoin transaction is not of type `ethtypes.Transaction`. In the `VerifyOutTxTrackerProof` function, we can then check the deserialization/unmarshal thing and verify `msg` against the original tx (eth or btc).

Reply:
okay makes sense , but in this PR , I am not dealing with BTC at all , we can do the refactor when merging you btc merkle proof pr ?

Reply:
I can help with the code unification when you sync with develop branch.